src/components/app/UrlManager.js

```diff
@@ -212,7 +212,10 @@ class UrlManagerImpl extends React.PureComponent<Props> {
       if (!getIsHistoryReplaceState()) {
         // Push the URL state only when the url setup is done, and we haven't set
         // a flag to only replace the state.
-        window.history.pushState(urlState, document.title, newUrl);
+        const result = newUrl.split('&');
+        if (result[1] !== 'v=5') {
```

Comment:
Unfortunately, that doesn't look correct. We have `v=5` in the url to indicate the url version. See:
https://github.com/firefox-devtools/profiler/blob/b3143cfdd9f771fb63b00abeb8e638f2f43c2556/src/app-logic/url-handling.js#L38
It's possible to change in the future since we are incrementing that whenever we have a change in the url strucure. So this will not work when we upgrade the url. Also checking `v=5` doesn't seem like the correct way to fix the solution even though the url version was going to stay 5 all the time. Another "valid" url can very well have the same string. Instead of doing this, we should maybe try to figure out "when" and "where" we are adding this invalid url to the history. So maybe we can eliminate updating the history overall.

Reply:
oh, alright. @canova I'll take a look at it again and implement your suggestions. Thank you for being patient with me.